### PR TITLE
Add timestamp filter

### PR DESCRIPTION
**Pull request by XAMPPRocky**
This adds a new timestamp filter that reads in a metadata value as a unix timestamp, and then observes the duration between then and now in a histogram metric that's keyed by the direction and metadata key. This is mainly useful in combination with things like `Capture` which then allows you to parse timestamps from packets.

### CODE REVIEW

**Review by markmandel — COMMENTED** [review]
This is awesome 😃 only thing missing is some docs please 👍🏻

File: src/filters/timestamp.rs [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+                return;
+            }
+        };
+
+        // Create a normal DateTime from the NaiveDateTime
+        let datetime: DateTime<Utc> = DateTime::from_utc(naive, Utc);
+
+        let now = Utc::now();
+        let seconds = now.signed_duration_since(datetime).num_seconds();
+        self.metric(direction_label).observe(seconds as f64);
```
Comment (markmandel) [inline src/filters/timestamp.rs:96]: (non blocking question): Does it make sense to combine this with maxmind information somehow? Either in the label or in tracing information?

Just thinking about tracking ping round trip times using this filter, and wanting some granularity on where geographically people are at?

**Review by markmandel — APPROVED** [review]

File: src/filters/timestamp.rs (hunk 1 above)
Comment (XAMPPRocky) [reply src/filters/timestamp.rs:96]: > Does it make sense to combine this with maxmind information somehow? Either in the label or in tracing information?

Yes it does, I'm going to try to find a way to do purely through grafana before trying to integrate it directly in code.

Comment (markmandel) [reply src/filters/timestamp.rs:96]: Make sense! 👍🏻

**Pull request merged**